packages/fuels-signers/src/wallet.rs

```diff
@@ -391,35 +567,92 @@ impl WalletUnlocked {
             .await?;
         let outputs = self.get_asset_outputs_for_amount(to, asset_id, amount);
 
-        // Build transaction and sign it
-        let mut tx = self
-            .get_provider()?
-            .build_transfer_tx(&inputs, &outputs, tx_parameters);
+        let mut tx = Wallet::build_transfer_tx(&inputs, &outputs, tx_parameters);
+
+        if asset_id == AssetId::default() {
+            self.add_fee_coins(&mut tx, amount, 0).await?;
+        } else {
+            self.add_fee_coins(&mut tx, 0, 0).await?;
+        };
```

Comment:
Can we add a comment here explaining why we decide between `amount` and `0`? Very confusing without proper context.